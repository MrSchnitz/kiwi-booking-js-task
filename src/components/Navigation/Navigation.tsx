import React from "react";
import {
  ButtonLink,
  Heading,
  NavigationBar,
  Stack,
} from "@kiwicom/orbit-components";
import { Link } from "react-router-dom";

const Navigation = React.memo(() => {
  return (
    <NavigationBar>
      <Stack direction="row" align="center" justify="start">
        <Link to="/">
          <ButtonLink>
            <img
              src="https://images.kiwi.com/common/kiwicom-logo.svg"
              alt="Kiwi.com logo"
            />
          </ButtonLink>
        </Link>
        <Heading type={"title1"}>Kiwi search</Heading>
      </Stack>
    </NavigationBar>
  );
});

export default Navigation;
