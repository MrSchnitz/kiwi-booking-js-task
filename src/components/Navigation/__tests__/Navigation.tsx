import { act, fireEvent, render } from "@testing-library/react";
import {
  MemoryRouter,
  unstable_HistoryRouter as HistoryRouter,
} from "react-router-dom";
import Navigation from "../Navigation";
import { createMemoryHistory } from "history";

describe("Navigation", () => {
  it("should render basic component", () => {
    const { container } = render(
      <MemoryRouter>
        <Navigation />
      </MemoryRouter>
    );

    expect(container).toMatchSnapshot();
  });

  it("should route to default route", () => {
    const history = createMemoryHistory({
      initialEntries: [{ pathname: "/about" }],
    });

    const { container } = render(
      <HistoryRouter history={history}>
        <Navigation />
      </HistoryRouter>
    );

    const linkButton: HTMLAnchorElement | null =
      container.querySelector('a[href="/"]');
    if (linkButton) {
      fireEvent.click(linkButton);
    }

    expect(history.location.pathname).toEqual("/");
  });
});
