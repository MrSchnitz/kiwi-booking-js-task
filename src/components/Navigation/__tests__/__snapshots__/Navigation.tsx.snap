// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation should render basic component 1`] = `
<div>
  <nav
    class="NavigationBar__StyledNavigationBar-sc-xlfrap-1 cCYMaj"
  >
    <div
      class="NavigationBar__StyledNavigationBarContent-sc-xlfrap-0 ffNPfp"
    >
      <div
        class="Stack__StyledStack-sc-oaff2v-0 iYCnXm"
      >
        <a
          href="/"
        >
          <button
            class="ButtonPrimitive__StyledButtonPrimitive-sc-1lbd19y-0 cEBSfu"
            type="button"
          >
            <div
              class="ButtonPrimitiveContent__StyledButtonPrimitiveContent-sc-1r81o9a-0 brimGn"
            >
              <div
                class="ButtonPrimitiveContentChildren__StyledButtonPrimitiveContentChildren-sc-1m4y8u8-0 jaHPWv"
              >
                <img
                  alt="Kiwi.com logo"
                  src="https://images.kiwi.com/common/kiwicom-logo.svg"
                />
              </div>
            </div>
          </button>
        </a>
        <div
          class="Heading__StyledHeading-sc-rrd5d-0 lfBIep"
        >
          Kiwi search
        </div>
      </div>
    </div>
  </nav>
</div>
`;
